bluetooth_mesh/cli/gatt_client.py

```diff
@@ -100,7 +102,27 @@ def _mesh_beacon_received(self, _, payload):
             self._network.beacon_receive(beacon, auth)
 
     def _network_pdu_received(self, _, payload):
-        pass
+        iv_index, ctl, ttl, seq, src, dst, transport_pdu= NetworkMessage.unpack(payload, 
+                                        self._network.network_keys, self._network.iv_index)
+        if transport_pdu:
+            if src >= 0x8000 or dst == 0:
+                return
+            for node in self._network.nodes.values():
+                for ele in node.elements:
+                    if ele['unicastAddress'] == src:
+                        if (iv_index << 24 | seq) <= (self._network.iv_index << 24 | ele["sequenceNumber"]):
+                            return
+                        ele["sequenceNumber"] = seq
+            key = (src, dst, seq)
+            if key in self._access_message_rxed.keys():
+                msg = self._access_message_rxed[key]
+            else:
+                msg = self._access_message_rxed[key] = AccessMessage(src, dst, ttl, None)
+            if not ctl:
+                decrypted_tp_pdu = msg.unpack(seq, iv_index, transport_pdu, self._network.application_keys)
+                if decrypted_tp_pdu and decrypted_tp_pdu != -1:
```

Comment:
Why `-1`?

Reply:
return a `None` from `msg.unpack` means reassembly is ongoing, `-1` means decryption failed.
in all wrong cases the message stored in `_access_message_rxed` should be discarded.

Reply:
I'd rather raise an exception if decryption fails. `-1` seems "too magic"

Reply:
yes, you are right.